flytekit/loggers.py

```diff
@@ -98,6 +93,10 @@ def initialize_global_loggers():
     set_flytekit_log_properties(handler, None, _get_env_logging_level())
     set_user_logger_properties(handler, None, logging.INFO)
 
+    # Use Rich logging while running in the local execution
+    if os.environ.get("FLYTE_INTERNAL_EXECUTION_ID", None) is None or interactive.ipython_check():
+        upgrade_to_rich_logging()
```

Comment:
For local execution, is it okay to initialize two loggers?

Reply:
What do you mean? In flytekit, we only have one global logger

Reply:
If `upgrade_to_rich_logging` is called, then I think there are two handlers. One `StreamHandler` from a few lines above and a `RichHandler` in `upgrade_to_rich_logging`.